src/backend_model.h

```diff
@@ -99,10 +114,18 @@ class TritonModel : public Model {
       const std::shared_ptr<LocalizedPath>& localized_model_dir,
       const std::shared_ptr<TritonBackend>& backend,
       const double min_compute_capability, const int64_t version,
-      const inference::ModelConfig& config, const bool auto_complete_config);
+      const inference::ModelConfig& config, const bool auto_complete_config,
+      const triton::common::BackendCmdlineConfigMap& backend_cmdline_config_map,
+      const triton::common::HostPolicyCmdlineConfigMap& host_policy_map);
 
-  // Set the scheduler based on the model configuration. The scheduler
-  // can only be set once for a backend.
+  // Replace the foreground instances with background instances.
```

Comment:
Actually, the background instance is for holding instances that will become foreground later. An instance can exist in both foreground and background at the same time, when it is about to be reused during an update. I think [this comment](https://github.com/triton-inference-server/core/pull/189#discussion_r1170629174) also supplements this part.

Some example flow/interaction between foreground and background:

When the model is initially created:
  - The background instance is empty.
  - `SetInstance` cannot match any foreground instance, because foreground is also empty.
  - All requested instances are created, into the background.
  - Commit the instances, so they are swapped into foreground (background <-- swap --> foreground).
  - All background instances are removed. (empty in this case)
  - Load complete.

An instance is removed during an update:
  - `SetInstance` is able to match all requested instance in foreground (all existing instances except the removed one as it is not requested). so the shared pointers are copied into background.
  - Commit the instances, so they are swapped into foreground (background <-- swap --> foreground).
  - All background instances are removed. The removed instance only has one reference count and it is in the background, so it is removed. The non-removed instances have reference count in the foreground, so they are not removed.
  - Update complete.

An instance is added during an update:
  - `SetInstance` is able to match all requested instance in foreground (all existing instances). so the shared pointers are copied into background.
  - `SetInstance` cannot find the added instance, so it is created into the background.
  - Commit the instances, so they are swapped into foreground (background <-- swap --> foreground).
  - All background instances are removed. All instances (except the added one) are in the background, but they have reference count in the foreground, so they are not removed. (no instances are removed in fact)
  - Update complete.